rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/vulnerability/Sw360VulnerabilityService.java

```diff
@@ -58,8 +69,16 @@ public Vulnerability getVulnerabilityByExternalId(String externalId, User sw360U
         try {
             VulnerabilityService.Iface sw360VulnerabilityClient = getThriftVulnerabilityClient();
             return sw360VulnerabilityClient.getVulnerabilityByExternalId(externalId, sw360User);
-        } catch (TException e) {
-            throw new RuntimeException(e);
+        } catch (SW360Exception sw360Exp) {
+            if (sw360Exp.getErrorCode() == 404) {
```

Comment:
Just a suggestion may be we can read these status code values from some constants file like SW360Constants, instead of hardcoding.

Reply:
Changed